arcaflow_plugin_pcp/pcp_plugin.py

```diff
@@ -141,109 +143,136 @@ def start_pcp(
                 "The pmlogger output file was not found; Unable to process results."
             )
 
-        pcp2_flags = [
-            "-a",
-            "pmlogger-out",
-            "-t",
-            str(params.pmlogger_interval),
-            "-f",
-            # pmrep doesn't accept %z for the timezone,
-            # so we'll get it explicitly via datetime
-            f"%FT%T.%f{str(datetime.now().astimezone())[-6:]}",
-            "-c",
-            pmrep_conf_path,
-            "-P",
-            "6",
-        ]
-
-        # Initialized pcp2json command
-        pcp2json_cmd = [
-            "/usr/bin/pcp2json",
-            "-E",
-        ]
-
-        pcp2json_cmd.extend(pcp2_flags)
+        post_process_params = {
+            "pmlogger_metrics": params.pmlogger_metrics,
+            "pmlogger_interval": params.pmlogger_interval,
+            "pmrep_conf_path": params.pmrep_conf_path,
+            "generate_csv": params.generate_csv,
+            "flatten": params.flatten,
+            "archive_path": ".",
+        }
+
+        return post_process(
+            params=post_process_params_schema.unserialize(post_process_params),
+            run_id="post-process",
```

Comment:
What's `run_id`? The `post_process` function defines a single parameter, `params`, so this should be inaccessible. Is there a point? Should it be taking a `**kwargs` and *using* this for something??? (Although the step decorator doesn't appear to allow that even if you wanted to... so I think this parameter is just silently thrown out, unless I'm missing something...)

Reply:
The `run_id` is a function of the SDK and is required.

Reply:
OK, I guess I didn't look deeply enough. I didn't see anything in the decorator that's actually *called* here that would care, but I suppose it's probably something else independently scavenging metadata.